In [1]:
import pandas as pd 
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import random

random.seed(100)

In [2]:
chipo = pd.read_table('chipotle.tsv.txt')

In [4]:
chipo.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

In [5]:
chipo.shape

(4622, 5)

there are 4622 observation

5 colums in the dataset

In [6]:
chipo.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

In [8]:
 chipo.tail()

order_id  quantity           item_name  \
4617      1833         1       Steak Burrito   
4618      1833         1       Steak Burrito   
4619      1834         1  Chicken Salad Bowl   
4620      1834         1  Chicken Salad Bowl   
4621      1834         1  Chicken Salad Bowl   

                                     choice_description item_price  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75

data is indexed from 0 to 4621

In [12]:
most_ordered = chipo.sort_values(by = 'quantity')

In [13]:
print(most_ordered)

      order_id  quantity                     item_name  \
0            1         1  Chips and Fresh Tomato Salsa   
3020      1201         1                   Canned Soda   
3021      1201         1                 Bottled Water   
3022      1201         1                 Side of Chips   
3023      1202         1                  Chicken Bowl   
...        ...       ...                           ...   
2441       970         5                 Bottled Water   
3599      1443         7                 Bottled Water   
3887      1559         8                 Side of Chips   
4152      1660        10                 Bottled Water   
3598      1443        15  Chips and Fresh Tomato Salsa   

                                     choice_description item_price  
0                                                   NaN     $2.39   
3020                                  [Diet Dr. Pepper]     $1.09   
3021                                                NaN     $1.09   
3022                       

chips and fresh tomato salsa is the most ordered


15 items were ordered

In [14]:
mostOrdered = chipo.sort_values(by = 'choice_description')

In [15]:
print(mostOrdered)

      order_id  quantity            item_name  \
510        214         1              Burrito   
675        279         1              Burrito   
521        217         1         Crispy Tacos   
673        279         1                 Bowl   
520        217         1         Crispy Tacos   
...        ...       ...                  ...   
4600      1827         1  Chips and Guacamole   
4605      1828         1  Chips and Guacamole   
4613      1831         1                Chips   
4614      1831         1        Bottled Water   
4616      1832         1  Chips and Guacamole   

                                     choice_description item_price  
510   [Adobo-Marinated and Grilled Chicken, Pinto Be...     $7.40   
675   [Adobo-Marinated and Grilled Chicken, [Sour Cr...     $7.40   
521               [Adobo-Marinated and Grilled Chicken]     $7.40   
673   [Adobo-Marinated and Grilled Steak, [Sour Crea...     $7.40   
520                 [Adobo-Marinated and Grilled Steak]     $7.40 

none was most ordered by this description

In [16]:
chipo['quantity'].sum()

4972

4972 items were ordered

item_price is already as float

In [19]:
chipo['item_price'].dtype

dtype('O')

In [24]:
chipos = chipo['item_price'].apply(lambda x: int(x) if isinstance(x, float) else x)

In [25]:
print(chipos)

0        $2.39 
1        $3.39 
2        $3.39 
3        $2.39 
4       $16.98 
         ...   
4617    $11.75 
4618    $11.75 
4619    $11.25 
4620     $8.75 
4621     $8.75 
Name: item_price, Length: 4622, dtype: object


In [33]:
chipos.dtypes

dtype('O')

In [36]:
a = chipo['item_price'].values.tolist()

In [42]:
chipo['item_name'].values.tolist()

['Chips and Fresh Tomato Salsa',
 'Izze',
 'Nantucket Nectar',
 'Chips and Tomatillo-Green Chili Salsa',
 'Chicken Bowl',
 'Chicken Bowl',
 'Side of Chips',
 'Steak Burrito',
 'Steak Soft Tacos',
 'Steak Burrito',
 'Chips and Guacamole',
 'Chicken Crispy Tacos',
 'Chicken Soft Tacos',
 'Chicken Bowl',
 'Chips and Guacamole',
 'Chips and Tomatillo-Green Chili Salsa',
 'Chicken Burrito',
 'Chicken Burrito',
 'Canned Soda',
 'Chicken Bowl',
 'Chips and Guacamole',
 'Barbacoa Burrito',
 'Nantucket Nectar',
 'Chicken Burrito',
 'Izze',
 'Chips and Fresh Tomato Salsa',
 'Chicken Bowl',
 'Carnitas Burrito',
 'Canned Soda',
 'Chicken Burrito',
 'Chips and Tomatillo-Green Chili Salsa',
 'Steak Burrito',
 'Side of Chips',
 'Carnitas Bowl',
 'Bottled Water',
 'Chicken Soft Tacos',
 'Chicken Soft Tacos',
 'Chips and Guacamole',
 'Chips and Tomatillo Green Chili Salsa',
 'Barbacoa Bowl',
 'Chips',
 'Chips and Guacamole',
 'Chicken Bowl',
 'Steak Burrito',
 'Chicken Salad Bowl',
 'Chicken Burrito',


In [43]:
chipo.shape

(4622, 5)

4633 orders

In [ ]:
import math

list_of_floats = a
list_of_floats = list(map(float, a))

total = math .fsum(a)